cmd/machine-api-operator/start.go

```diff
@@ -159,3 +171,30 @@ func startHTTPMetricServer(metricsPort string) {
 	}
 	glog.Fatal(server.ListenAndServe())
 }
+
+// CreateResourceLock returns an interface for the resource lock.
+func CreateResourceLock(cb *ClientBuilder, componentNamespace, componentName string) resourcelock.Interface {
+	recorder := record.
+		NewBroadcaster().
+		NewRecorder(scheme.Scheme, v1.EventSource{Component: componentName})
+
+	id, err := os.Hostname()
+	if err != nil {
+		glog.Fatalf("Error creating lock: %v", err)
+	}
+
+	// add a uniquifier so that two processes on the same host don't accidentally both become active
+	id = id + "_" + string(uuid.NewUUID())
+
+	return &resourcelock.ConfigMapLock{
+		ConfigMapMeta: metav1.ObjectMeta{
+			Namespace: componentNamespace,
+			Name:      componentName,
+		},
+		Client: cb.KubeClientOrDie("leader-election").CoreV1(),
```

Comment:
Return an error? It makes unit testing easier.

Reply:
The second commit is just moving code as there's no need to have a helpers file for this function. Let's keep this PR for changing the default resync and I'll put any farther improvement in a different one